Sources/AWSSDKSwiftCore/Message/Body.swift

```diff
@@ -55,53 +55,47 @@ extension Body {
             return false
         }
     }
-    
+
     public func asDictionary() throws -> [String: Any]? {
         switch self {
-            
+
         case .json(let data):
             return try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
-            
+
         default:
             return nil
         }
     }
-    
+
     public func asData() throws -> Data? {
         switch self {
         case .text(let text):
             return text.data(using: .utf8)
-            
+
         case .buffer(let data):
             return data
-            
+
         case .json(let data):
             if data.isEmpty {
                 return nil
             } else {
                 return data
             }
-            
+
         case .xml(let node):
-            if let document = node as? XMLDocument {
-                return document.xmlData
-            } else if let element = node as? XMLElement {
-                let xmlDocument = XMLDocument(rootElement: element)
-                xmlDocument.version = "1.0"
-                xmlDocument.characterEncoding = "UTF-8"
-                return xmlDocument.xmlData
-            }
-            return nil
-            
+            let xmlDocument = XML.Document(rootElement: node)
+            xmlDocument.version = "1.0"
```

Comment:
Could/should those be set in the initializer if not passed through by default? 🤔

Reply:
My initial implementation was designed to stick to the Foundation implementation as tight as possible so moving between the two would be easy as possible. This is how it is implemented in the Foundation. I have already diverged slightly from the Foundation API, so can do so more.

Reply:
I removed these two lines as they aren't needed since the class output defaults to these values

Reply:
Awesome, thanks!